src/Apps/Sell/SubmissionContext.tsx

```diff
@@ -0,0 +1,162 @@
+import * as Yup from "yup"
+import { SubmissionContextUpdateSubmissionMutation } from "__generated__/SubmissionContextUpdateSubmissionMutation.graphql"
+import { createContext, useState } from "react";
+import { graphql } from "react-relay";
+import { isEqual } from "lodash";
+import { useMutation } from "Utils/Hooks/useMutation";
+import { useRouter } from "System/Router/useRouter";
+
+const TitleStepSchema = Yup.object().shape({
+  title: Yup.string().required().trim(),
+})
+
+const INPUT_SCHEMAS = {
+  "title": TitleStepSchema
+}
+
+interface Actions {
+  goBack: () => void
+  saveAndContinue: () => void
+  submitSubmission: () => void
+  atFirstStep: () => boolean
+  atLastStep: () => boolean
+  currentStep: () => string
+  isSaving: () => boolean
+  isValid: () => boolean
+  onChange: (event: any) => void
+  updatePayload: (input: {}) => void
+  setInitialValues: (values: any) => void
+  isDirty: () => boolean
+}
+
+interface State {
+  payload: any
+}
+
+interface SubmissionContextProps {
+  actions: Actions
+  state: State
+}
+
+export const SubmissionContext = createContext<SubmissionContextProps>({} as any)
+
+export const STEPS = [
+  "title",
+  "photos",
+  "details",
+  "purchase-history",
+  "dimensions"
+]
```

Comment:
I wonder if we could derive the step directly from the `pathname`? This way, we wouldn't need to sync the step in the context and the pathname 🤔